### PR TITLE
fix(sql): improved the duplicated column names message

### PR DESCRIPTION
**Pull request by kocko**

### CODE REVIEW

File: core/src/main/java/io/questdb/griffin/SqlParser.java [hunk 1]
```diff
@@ -561,6 +562,20 @@ private void parseCreateTableAsSelect(GenericLexer lexer, CreateTableModel model
         expectTok(lexer, ')');
     }
 
+    private void ensureThatColumnNamesAreUnique(ObjList<QueryColumn> columns) throws SqlException {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/SqlParser.java:565]: this should be done via `LowerCaseCharSequenceHashSet` to ensure:

- O(log n)
- case sensitivity is dealt with

Algorithm also needs to take into account column aliases where present

N.B. `LowerCaseCharSequenceHashSet` instance can be a class variable, which is reused and cleared after use

File: core/src/test/java/io/questdb/griffin/SqlParserTest.java [hunk 2]
```diff
@@ -640,6 +640,28 @@ public void testCreateAsSelectTimestampNotRequired() throws SqlException {
         );
     }
 
+    @Test
+    public void testCreateAsSelectDuplicatedColumnNames() throws Exception {
+        assertSyntaxError("create table all as (" +
+                "    select" +
```
Comment (bluestreak01) [inline core/src/test/java/io/questdb/griffin/SqlParserTest.java:646]: tests should also include mixed cases and non-ascii (supported) characters

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/griffin/SqlParser.java (hunk 1 above)
Comment (kocko) [reply core/src/main/java/io/questdb/griffin/SqlParser.java:568]: Improved the complexity and added tests. 

Column aliases will further be discussed.

**Comment by kocko** [comment]
When I execute `create table something as (select event ev, other ev from telemetry)` in PostgreSQL, I get an error, saying that the column `ev` already exists.

I now think that if similar behaviour is desired, then the `SqlUnit#createColumnAlias()` should not try to resolved name conflicts, but raise an error instead.

**Comment by bluestreak01** [comment]
Yes, agreed. Duplicate aliases should be producing `SqlException`

**Comment by kocko** [comment]
In addition,

- `select 2,2 from long_sequence(10);` shoult not produce any error;
- `create table temp (select 2, 2 from long_sequence(10)); should produce an error;
- `create table temp (select 2 two, 2 another_two from long_sequence(10));` should not produce an error.
- `create table temp (select 2 two, 2 two from long_sequence(10));`  should produce an error.

**Comment by bluestreak01** [comment]
stale PR

**Pull request closed without merging**